pennylane/tape/qscript.py

```diff
@@ -1410,3 +1411,68 @@ def wrapper(*args, **kwargs):
         return qscript
 
     return wrapper
+
+
+def bind_new_parameters_tape(
+    tape: QuantumScript, params: Sequence[TensorLike], indices: Sequence[int]
+):
+    """Create a new tape with updated parameters.
+
+    This function takes a :class:`~.tape.QuantumScript` as input, and returns
+    a new ``QuantumScript`` containing the new parameters at the provided indices,
+    with the parameters at all other indices remaining the same.
+
+    Args:
+        tape (.tape.QuantumScript): Tape to update
+        params (Sequence[TensorLike]): New parameters to create the tape with. This
+            must have the same length as ``indices``.
+        indices (Sequence[int]): The parameter indices to update with the given parameters.
+            The index of a parameter is defined as its index in ``tape.get_parameters()``.
+
+    Returns:
+        .tape.QuantumScript: New tape with updated parameters
+    """
+    # pylint: disable=no-member
+
+    if len(params) != len(indices):
+        raise ValueError("Number of provided parameters does not match number of indices")
+
+    # determine the ops that need to be updated
+    op_indices = {}
+    for param_idx, idx in enumerate(sorted(indices)):
+        pinfo = tape._par_info[idx]
+        op_idx, p_idx = pinfo["op_idx"], pinfo["p_idx"]
```

Comment:
```suggestion
        _, op_idx, p_idx = tape._par_info[idx]
```

Reply:
this won't work, `tape._par_info["idx"]` is a dict, and it has 3 keys

Reply:
Yep, this won't work.